src/index.ts

```diff
@@ -61,44 +60,50 @@ import {
   convertApiCountry,
   convertSdkPriceList,
 } from './helpers/converters'
+import { MetadataCodec } from './metadata'
 
 export * from './exceptions'
 export * from './interfaces'
-
-export interface PaddleSdkOptions {
-  publicKey: string
-  vendorId: number
-  vendorAuthCode: string
-  metadataEncryptionKey: string
+export * from './metadata'
+
+export interface PaddleSdkOptions<TMetadata = unknown> {
+  /** Public key from the paddle dashboard to validate webhook requests */
+  readonly publicKey: string
+  /** Vendor ID from the paddle dashboard to authenticate API requests */
+  readonly vendorId: number
+  /** Vendor auth code from the paddle dashboard to authenticate API requests */
+  readonly vendorAuthCode: string
+  /**
+   * Metadata codec encodes and decodes additional pass-through data for an order
+   *
+   * @see https://developer.paddle.com/guides/how-tos/checkout/pass-parameters
+   */
+  readonly metadataCodec: MetadataCodec<TMetadata>
```

Comment:
Tiny nitpick - add newlines

```suggestion
export interface PaddleSdkOptions<TMetadata = unknown> {
  /** Public key from the paddle dashboard to validate webhook requests */
  readonly publicKey: string

  /** Vendor ID from the paddle dashboard to authenticate API requests */
  readonly vendorId: number

  /** Vendor auth code from the paddle dashboard to authenticate API requests */
  readonly vendorAuthCode: string

  /**
   * Metadata codec encodes and decodes additional pass-through data for an order
   *
   * @see https://developer.paddle.com/guides/how-tos/checkout/pass-parameters
   */
  readonly metadataCodec: MetadataCodec<TMetadata>
```